### ISSUE TITLE
The `util.cos_sim` always uses `Tensor` object and also returns the same when we are working with numpy

### ISSUE THREAD

**Issue by AayushSameerShah** [UKPLab/sentence-transformers#2513]
I am running a simple sentence similarity code:

```python
from sentence_transformers import SentenceTransformer, util
sentence_transformers_model = SentenceTransformer("all-MiniLM-L6-v2")

## The code begins
emb1 = sentence_transformers_model.encode(corpus1, convert_to_numpy=True)
emb2 = sentence_transformers_model.encode(corpus2, convert_to_numpy=True)

## And the score for all pairs
cosine_scores = util.cos_sim(emb1, emb2)
```

> Now the thing is, I wanted to work with only numpy arrays, that's why initially I started with `convert_to_numpy`. But the `util.cos_sim` converts into `Tensor`.

I can always convert that tensor object back to numpy by `cosine_scores.cpu().numpy()` but that will be some conversion overhead.

Isn't there any `util.cos_sim` flag that returns the numpy array itself?
Thanks for the help 🙏🏻

**Comment by tomaarsen** [comment]
Hello!

I think it makes sense to try and preserve the object, i.e. if you provide a numpy array then it stays a numpy array, and the same for torch tensors. I'll try and include this on top of #2490 as that PR also affects these utility functions.

- Tom Aarsen

**Comment by AayushSameerShah** [comment]
Sounds good! Looking forward to this! 🤞🏻

**Comment by tomaarsen** [comment]
Hmm, upon further consideration, I do have to be careful with breaking changes: I can't update the existing `cos_sim` to return numpy arrays when given numpy arrays, as people's code might break as a result.

A flag might be a better solution.

- Tom Aarsen

**Comment by AayushSameerShah** [comment]
Yes! I agree. The flag would be the solution, and it also won't break the library style since we are already using such flags in other parts like `convert_to_numpy` 🤗
